### DISCUSSION TITLE
Somehow trigger tag creation event manually?

### DISCUSSION THREAD

**Discussion by jinnatar** [woodpecker-ci/woodpecker#4462] [Ask a Question]
Sometimes due to the position of the stars or whatever the webhook from the forge for tag creation just goes AWOL. I can't seem to find any graceful way to recover from this unless the forge has recorded the webhook failure and allows a replay of it. Sometimes that's an option, but this time today it's not.

Sample config:
```yaml
---
steps:
  publish:
    image: plugins/docker
    settings:
      repo: docker.example.com/o11n/sample_service
      registry: docker.example.com
      auto_tag: true
```

Ideas considered and tried:
1. Re-trigger the webhook that failed to go through from the forge. In this instance there was most likely a momentary DNS issue and for some reason Forgejo did not leave behind a failed webhook record, so nothing there to replay.
1. Trigger the webhook manually from the forge. The "test" button sends a "push" and there doesn't seem to be a way to get a specific type sent.
3. Create a pipeline manually in Woodpecker. Sadly neither the web UI or cli allows specifying which event to trigger, it's always `manual`. Setting branch to the tag value doesn't work, it tries to be too clever and expects the value to actually be a branch, not an arbitrary ref.
4. Do some wacky shenanigans in the repo CI config itself to route custom variables to also trigger a tag-like build and instead take the tag value from the variable? I guess it would work, but seems like a huge hack and complication to an otherwise oh so simple config. Autotagging is super nice, but only if the tag event is actually there to consume!
5. Forge the webhook myself manually. Very labor intensive and would need to MitM a webhook to even get started. I really hope this isn't the only option.
6. Make some arbitrary NOOP code change, force apply the same tag on the new revision and force push the tag again! Hurts my heart to even consider this, but I guess it would work.

Is there some simple option I'm not seeing? It just feels wrong that a single errant webhook call going missing completely makes it impossible to recover cleanly because there are no options to synthesize any event other than `manual`. That's likely a security consideration, but as said I could just go push an arbitrary webhook via curl given enough effort, so seems like there is perhaps less security value than seems at first.

Edit:
- One more option I found that at least allows one chance at recovery: Create a release out of the tag even if you don't normally do releases. That works with `auto_tag`. :woman_shrugging:

**Comment by zc-devs** [comment]
4 - I do this in general. Doesn't work with plugins.
```sh
    app_version=$(printf '%s' "$CI_COMMIT_SHA" | head -c 8)
    if [ -n "${CI_MANUAL_TAG-}" ]; then
        app_version="$CI_MANUAL_TAG"
    elif [ -n "${CI_COMMIT_TAG-}" ]; then
        app_version="$CI_COMMIT_TAG"
    fi
```
1 - And this also works sometimes.
6 - Is code change necessary? I use lightweight tags, so I managed just to delete the tag and create again on the same commit.
